examples/locality-aware-routing/README.md

```diff
@@ -0,0 +1,143 @@
+
+![Example architecture](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/_docs/locality-aware-routing-arch.png?raw=true)
+
+This terraform example deploys a client application and two server application tasks spread across available within the same AWS region. The consul server is also deployed as an ECS task along with an ECS controller.
+
+## Requirements
+
+* `jq`
+* `curl`
+* Terraform >= 1.2.2
+* Authentication credentials for the [Terraform AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication)
+
+## Usage
+
+### Setup
+
+Clone this repository:
+
+```console
+$ git clone https://github.com/hashicorp/terraform-aws-consul-ecs.git
+$ git checkout tags/<latest-version>
+$ cd terraform-aws-consul-ecs/examples/locality-aware-routing
+```
+
+This module contains everything needed to spin up the example. The only
+requirements are:
+- You need to pass in the `name` variable. This value will be used as a unique identifier
+  for all resources created by the example. The examples below use the name `ecs`.
+- You need to pass in the IP address of your workstation via the `lb_ingress_ip`
+  variable. This is used for the security groups on the elastic load balancers to ensure
+  only you have access to them.
+- Consul Enterprise license that needs to be passed to the `consul_license` variable. This license will be used to run the enterprise version of Consul in this example. One way to pass these licenses would be add them to a `.tfvars` file and pass it as an argument to the `terraform apply` command.
+
+Determine your public IP. You can use a site like https://ifconfig.me/:
+
+```console
+$ curl ifconfig.me
+123.456.789.1%
+```
+
+Example `input.tfvars`
+
+```
+lb_ingress_ip  = "123.456.789.1"
+consul_license = "<license>"
+```
+
+Initialize Terraform:
+
+```console
+$ terraform init
+```
+
+### Terraform Apply
+
+Then apply the Terraform passing in a name and your IP:
+
+```console
+$ terraform apply \
+    -var name=ecs \
+    -var-file=input.tfvars
+```
+
+The plan should look similar to:
+
+```shell
+Plan: 81 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + client_lb_address             = (known after apply)
+  + consul_server_bootstrap_token = (sensitive value)
+  + consul_server_url             = (known after apply)
+```
+
+Type `yes` to apply the changes.
+
+~> **Warning:** These resources will cost money. Be sure to run `terraform destroy`
+   when you've finished testing.
+
+The apply should take 7-10 minutes. When complete, the URLs of the two load
+balancers should be in the output, along with the bootstrap token for the Consul server:
+
+```shell
+Apply complete! Resources: 81 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+client_lb_address = "http://example-client-app-1959503271.us-west-2.elb.amazonaws.com:9090/ui"
+consul_server_bootstrap_token = <sensitive>
+consul_server_url = "http://ecs-dc1-consul-server-713584774.us-west-2.elb.amazonaws.com:8500"
+```
```

Comment:
I got a bunch of permission denied errors when I tried this in my dev account. I'm wondering if it is a problem with my account or there are policies missing. I've never seen this type of thing before.

```
│ Error: error creating Secrets Manager Secret: AccessDeniedException: User: arn:aws:sts::xxxxxxxxxxxx:assumed-role/aws_chris.thain_test-developer/chris.thain@hashicorp.com is not authorized to perform: secretsmanager:CreateSecret on resource: ecs-dc1-dc1-bootstrap-token with an explicit deny in a session policy
```

Reply:
I get such errors when I connect my Mac using my phone as the Wifi provider. I don't see the same when I connect it directly to my modem. Not sure why that's the case though